front/components/skill_builder/submitSkillBuilderForm.ts

```diff
@@ -1,81 +1,116 @@
 import type { Result, UserType, WorkspaceType } from "@app/types";
 import { Err, Ok } from "@app/types";
 
 export async function submitSkillBuilderForm({
   formData,
   owner,
   user,
+  skillConfigurationId,
 }: {
   formData: SkillBuilderFormData;
   owner: WorkspaceType;
   user: UserType;
+  skillConfigurationId?: string;
 }): Promise<
-  Result<PostSkillConfigurationResponseBody["skillConfiguration"], Error>
+  Result<
+    | PostSkillConfigurationResponseBody["skillConfiguration"]
+    | PatchSkillConfigurationResponseBody["skillConfiguration"],
+    Error
+  >
 > {
-  const response = await clientFetch(
-    `/api/w/${owner.sId}/assistant/skill_configurations`,
-    {
-      method: "POST",
+  try {
+    const endpoint = skillConfigurationId
+      ? `/api/w/${owner.sId}/assistant/skill_configurations/${skillConfigurationId}`
+      : `/api/w/${owner.sId}/assistant/skill_configurations`;
+
+    const method = skillConfigurationId ? "PATCH" : "POST";
+
+    const response = await clientFetch(endpoint, {
+      method,
       headers: {
         "Content-Type": "application/json",
       },
       body: JSON.stringify({
         name: formData.name,
         description: formData.description,
         instructions: formData.instructions,
-        scope: formData.scope,
         tools: formData.tools.map((tool) => ({
           mcpServerViewId: tool.configuration.mcpServerViewId,
         })),
       }),
+    });
+
+    if (!response.ok) {
+      const errorData = await response.json();
+      return new Err(
+        new Error(
+          errorData.error?.message ??
+            (skillConfigurationId
+              ? "Failed to update skill"
+              : "Failed to create skill")
+        )
+      );
     }
-  );
 
-  if (!response.ok) {
-    const errorData = await response.json();
-    return new Err(
-      new Error(errorData.error?.message ?? "Failed to create skill")
-    );
-  }
+    const result:
+      | PostSkillConfigurationResponseBody
+      | PatchSkillConfigurationResponseBody = await response.json();
 
-  const result: PostSkillConfigurationResponseBody = await response.json();
+    const skillConfiguration = result.skillConfiguration;
 
-  const skillConfiguration = result.skillConfiguration;
+    const desiredEditorIds = new Set(formData.editors.map((e) => e.sId));
+    const creatorId = user.sId;
 
-  const desiredEditorIds = new Set(formData.editors.map((e) => e.sId));
-  const creatorId = user.sId;
+    const addEditorIds: string[] = [];
+    const removeEditorIds: string[] = [];
 
-  const addEditorIds: string[] = [];
-  const removeEditorIds: string[] = [];
+    for (const editor of formData.editors) {
+      if (editor.sId !== creatorId) {
+        addEditorIds.push(editor.sId);
+      }
+    }
 
-  for (const editor of formData.editors) {
-    if (editor.sId !== creatorId) {
-      addEditorIds.push(editor.sId);
+    if (!desiredEditorIds.has(creatorId)) {
+      removeEditorIds.push(creatorId);
```

Comment:
I might be missing something here, but we ever only remove the "creatorId", not any other user